sidechain/state/src/lib.rs

```diff
@@ -203,4 +223,10 @@ impl<T: SidechainState> SidechainSystemExt for T {
 	fn set_timestamp(&mut self, timestamp: &Timestamp) {
 		self.set_with_name("System", "Timestamp", timestamp)
 	}
+
+	fn reset_events(&mut self) {
```

Comment:
@clangenb @mullefel  This is not yet a review Request, but a design question.

We have our `sidechain/state` file but also our `app-libs/stf`. In both I have defined a `reset_events` function. Because to my understanding, `sidechain` and `app-libs` must not depend on each other.

However, I don't like the current design where we have relatively many code duplications for state access. Do you have a future plan in mind,  how to change this, or did I overlook something?

Reply:
It's not a specific question about `reset_events` but general duplicated functions in these crates.

Reply:
I think it makes sense to move some helper functions to itp_storage. Suggestion incoming as PR.

Reply:
I think we should in the future extract some implementations on the `STF` to `core/stf-system` analogous to `frame-system` or something of the like (naming to be discussed, I am uncertain if it makes sense to stay so close to substrate).

Reply:
Yeah, I think that would make sense.

Reply:
Should be resolved with https://github.com/integritee-network/worker/issues/1036